### PR TITLE
Add "Proxy" to terminology page

### LINKED ISSUE #3086: add "proxy" to terminology list
### Problem you are trying to solve

I'm reading [this](https://rust-lang.github.io/rustup/concepts/index.html) page, the list of terminology. I see that "proxy" is missing. It should be mentioned in the terminology list. 

### Solution you'd like

add "proxy" to the terminology list, along with brief description

### Notes

_No response_

### PR DESCRIPTION
**Pull request by x3ro**
Fixes #3086

Hope this is roughly what was expected 😅

### CODE REVIEW

File: doc/src/concepts/index.md [hunk 1]
```diff
@@ -48,9 +48,13 @@ This is similar to Ruby's [rbenv], Python's [pyenv], or Node's [nvm].
   "profile" defines a grouping of components. See the [Profiles] chapter for
   more details.
 
+* **proxy** —  A wrapper for a common Rust component (e.g. `rustc`), built to forward
```
Comment (0xPoe) [inline doc/src/concepts/index.md]:
```suggestion
* **proxy** —  A wrapper for a common Rust component (like `rustc`), built to forward
```
Keep the same style as above.

**Review by 0xPoe — COMMENTED** [review]
Thanks!

File: doc/src/concepts/index.md (hunk 1 above)
Comment (ghost) [reply doc/src/concepts/index.md]: Done :)

**Review by 0xPoe — APPROVED** [review]
Thanks!

**Pull request merged**